rpc/core/pipe.go

```diff
@@ -44,7 +44,8 @@ type transport interface {
 }
 
 type peers interface {
-	DialPeersAsync(p2p.AddrBook, []string, bool) error
+	AddPersistentPeers([]string) []error
```

Comment:
Everywhere this is used, only the first error `errs[0]` is considered. Why return a list of errors then?

Reply:
Because someone (external users of our p2p lib) could need all the errors?

Reply:
Not sure if there are users of the p2p library. But it is likely that they will use it as it we use it internally too: probably, they will only check "was there an error" (and use the first).

If it is important that the interface method takes several addrs, then only one `error`(the first occurring?) should be retuned. This reflects the current implementation. If you think people must have the possibility to treat the errors separately, why not have a method that takes one addr string and returns one error instead?

Reply:
> probably, they will only check "was there an error"

I wound not make such statements.

> why not have a method that takes one addr string and returns one error instead?

being able to add multiple addresses is more convenient, I guess.

Reply:
> I wound not make such statements.

Sorry, you are right. It's far too vague and sounds like wild guessing. What I'm trying to say, is that in golang it is very idiomatic that a method returns one error if sth. went wrong. Ideally, in some cases it returns a set of known errors users can switch over and handle in a particular way. Returning a list of errors is less idiomatic. Hence, it isn't unlikely that a user will treat the list of errors like "did an error occur" (in fact your code even does that).

> being able to add multiple addresses is more convenient, I guess.

Absolutely! But it obfuscates the connection between addr <-> potentially occurred error. Letting the caller use a for-loop and range over the addresses (and deal with those errors), leaves no doubt about this connection.

Reply:
> What I'm trying to say, is that in golang it is very idiomatic that a method returns one error

good argument